lib/hipmer/hipmerUtils.py

```diff
@@ -448,12 +448,27 @@ def finish_run(self, params):
             raise RuntimeError("Error in HipMER execution")
 
         wsname = params['workspace_name']
-        self.log(console, 'Uploading FASTA file to Assembly')
+
+        self.log(console, 'Filtering short length contigs from HipMer assembly')
+
         assemblyUtil = AssemblyUtil(self.callbackURL, token=self.token)
+
+        assembly_size_filter = params['assembly_size_filter']
+
+        filtered_fasta_file_path = assemblyUtil.filter_contigs_by_length(output_contigs, assembly_size_filter)
+
+        if os.stat(filtered_fasta_file_path).st_size == 0:
```

Comment:
Hm. Would it be better to just fail? If all the contigs are smaller than the user wants, giving them a ton of tiny contigs seems like the wrong thing to do.

Reply:
Good point. I opted for this route to minimize the chances that a re-run is needed.

Reply:
Yeah.  I was unsure about this too.  But this may be fine.

Reply:
> Good point. I opted for this route to minimize the chances that a re-run is needed.

If a user doesn't want contigs smaller than X, but the results contain contigs smaller than X, then a re-run is required anyway. Even worse, the user may not notice that their input was ignored and continue with, essentially, corrupted data.

Reply:
Not sure I follow. This is filtering the output assembly, not the user's input data.

Reply:
Right. The user is asking for an output assembly with no contigs < X size. The app may give them contigs < X size, so it's not following the user's instructions. Worse, if the user doesn't notice the app didn't follow the user's instructions, they could keep going with their analysis on invalid data.

Reply:
Okay, think I'm clear. So you are advocating the App just fails in these scenarios. I'm not against this since those cases should be exceedingly rare. Was trying to save compute in this unlikely scenario, but perhaps that's not the best way.

Reply:
This looks better to me. Let me know if there's a better error to raise